priv/static/phoenix_html.js

```diff
@@ -1,41 +1,37 @@
-'use strict';
+"use strict";
 
-function isLinkToSubmitParent(element) {
-  var isLinkTag = element.tagName === 'A';
-  var shouldSubmitParent = element.getAttribute('data-submit') === 'parent';
-
-  return isLinkTag && shouldSubmitParent;
-}
-
-function getClosestForm(element) {
-  while (element && element !== document && element.nodeType === Node.ELEMENT_NODE) {
-    if (element.tagName === 'FORM') {
-      return element;
-    }
-    element = element.parentNode;
+(function() {
+  function buildHiddenInput(name, value) {
+    var input = document.createElement("input");
+    input.type = "hidden";
+    input.name = name;
+    input.value = value;
+    return input;
   }
-  return null;
-}
 
-function didHandleSubmitLinkClick(element) {
-  while (element && element.getAttribute) {
-    if (isLinkToSubmitParent(element)) {
-      var message = element.getAttribute('data-confirm');
-      if (message === null || confirm(message)) {
-        getClosestForm(element).submit();
-      }
-      return true;
-    } else {
-      element = element.parentNode;
+  function handleLinkClick(link) {
+    if(link.dataset["confirm"] == "true" && !window.confirm("Are you sure?")) {
```

Comment:
I guess for this we could stay with the previous behavior, don't we?

It's better if we keep the message on the HTML because then we can use `<%= gettext("Are you sure?") %>` for translated messages.